contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol

```diff
@@ -153,7 +153,14 @@ abstract contract FunctionsSubscriptions is IFunctionsSubscriptions, IERC677Rece
   function ownerCancelSubscription(uint64 subscriptionId) external override {
     _onlyRouterOwner();
     _isExistingSubscription(subscriptionId);
-    _cancelSubscriptionHelper(subscriptionId, s_subscriptions[subscriptionId].owner);
+
+    (uint96 balance, , address subscriptionOwner) = _cancelSubscriptionHelper(subscriptionId);
```

Comment:
Would it help if you put all the logic inside the helper but pass a flag indicating whether to withhold the deposit or not?

Reply:
Maybe. Not sure if it will get us all the way there, but I'll give it a try!

Reply:
-259 bytes. Helps, but not quite there!